File: examples/go/logs.go
// Example showing how to place trades, check results, and log activity
package main

import (
	"fmt"
	"time"
	"binary_options_tools_uni"
)

func main() {
	client, err := binary_options_tools_uni.NewPocketOption("your-session-id")
	if err != nil {
		panic(err)
	}
	time.Sleep(5 * time.Second)

	fmt.Println("Starting trading operations...")

	// Place a buy (call) trade
	buyDeal, err := client.Buy("EURUSD_otc", 300, 1.0)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Buy trade placed with ID: %s\n", buyDeal.ID)

	// Place a sell (put) trade
	sellDeal, err := client.Sell("EURUSD_otc", 300, 1.0)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Sell trade placed with ID: %s\n", sellDeal.ID)

	fmt.Println("\nWaiting for trades to complete (65 seconds)...")
	time.Sleep(65 * time.Second)

	// Check results for both trades
	buyResult, err := client.CheckWin(buyDeal.ID)
	if err != nil {
		panic(err)
	}
	fmt.Printf("\nBuy trade result: %+v\n", buyResult)

	sellResult, err := client.CheckWin(sellDeal.ID)
	if err != nil {
		panic(err)
	}
	fmt.Printf("Sell trade result: %+v\n", sellResult)
}

File: examples/ruby/logs.rb
# Start logging, place trades, and check results example
require_relative 'binary_options_tools_uni'

client = BinaryOptionsToolsUni::PocketOption.new("your-session-id")
sleep 5

balance_before = client.balance
puts "Balance before: $#{balance_before}"

# Place a buy trade
deal = client.buy("EURUSD_otc", 300, 1.0)
puts "Buy trade placed with ID: #{deal.id}"

# Place a sell trade
sell_deal = client.sell("EURUSD_otc", 300, 1.0)
puts "Sell trade placed with ID: #{sell_deal.id}"

puts "Waiting for trades to complete..."
sleep 305

# Check results of both trades
buy_result = client.check_win(deal.id)
puts "Buy trade result: #{buy_result}"

sell_result = client.check_win(sell_deal.id)
puts "Sell trade result: #{sell_result}"

balance_after = client.balance
puts "Balance after: $#{balance_after}"
puts "Profit/Loss: $#{balance_after - balance_before}"
